spec-tests/src/test/java/org/candlepin/spec/content/ContentVersioningSpecTest.java

```diff
@@ -0,0 +1,314 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.content;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertNotFound;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import org.candlepin.dto.api.client.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.client.v1.ContentDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Content;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@SpecTest
+public class ContentVersioningSpecTest {
+    private ApiClient adminClient;
+    private OwnerDTO owner;
+
+    @BeforeEach
+    public void beforeEach() {
+        adminClient = ApiClients.admin();
+        owner = adminClient.owners().createOwner(Owners.random());
+    }
+
+    @Test
+    public void shouldCreateOneContentInstanceWhenSharedByMultipleOrgs() throws Exception {
+        OwnerDTO owner2 = adminClient.owners().createOwner(Owners.random());
+        ContentDTO expectedContent = Content.random();
+
+        ContentDTO owner1Content = adminClient.ownerContent().createContent(owner.getKey(), expectedContent);
+        ContentDTO owner2Content = adminClient.ownerContent().createContent(owner2.getKey(), expectedContent);
+
+        assertThat(owner1Content)
```

Comment:
We're missing the most critical aspect of this test: verifying that the UUIDs on both content is the same. I know this is probably covered by the `isEqualTo` check, but given that it's the core component here, I would expect it to be done explicitly.

Reply:
That's a good point, we should explicitly verify that the UUIDs are the same and not rely on the `isEqualTo` check. Updated